Framework/Core/src/ExternalFairMQDeviceProxy.cxx

```diff
@@ -41,7 +41,33 @@ void sendOnChannel(FairMQDevice& device, FairMQParts& messages, std::string cons
   // array of channels, the index is 0 in the call
   constexpr auto index = 0;
   LOG(DEBUG) << "sending " << messages.Size() << " messages on " << channel;
-  device.Send(messages, channel, index);
+  // TODO: we can make this configurable
+  const int maxTimeout = 10000;
+  int timeout = 0;
+  // try dispatch with increasing timeout in order to also drop a warning if the dispatching
+  // has been tried multiple times within max timeout
+  // since we do not want any messages to be dropped at this stage, we stay in the loop until
+  // the downstream congestion is resolved
+  // TODO: we might want to treat this error condition some levels higher up, but for
+  // the moment its an appropriate solution. The important thing is not to drop
+  // messages and to be informed about the congestion.
+  while (device.Send(messages, channel, index, timeout) < 0) {
+    if (timeout == 0) {
+      timeout = 1;
+    } else if (timeout < maxTimeout) {
+      timeout *= 10;
+    } else {
+      LOG(ERROR) << "failed to dispatch messages on channel " << channel << ", downstream queue might be full\n"
```

Comment:
I'm not sure if the message is clear enough. This has potential to drop parts of TimeFrame from reaching processing components and potentially causing a lot of confusion. Effect of this in fact will relax back-pressure for upstream components.

This is a clearly an error condition and the usability of results is void if we're drop data like this. Perhaps increasing the `maxTimeout` and stating it clearly that this is most likely a BUG.
What do you think?

Reply:
@mkrzewic What do you think?

Reply:
At this point we do not drop data, we simply keep on trying, so back pressure should not be relaxed upstream. In order to know what is going on, we need some error message. How would you formulate it?

Its not necessarily a bug, well depending on the definition of _bug_, can be a slow device downstream and resulting congestion.

Reply:
we need to add the information that no data is dropped to the message, agreed

Reply:
Sorry, my bad. This will keep trying even after maxTimeout.

Can there be information about where in the workflow this is happening? To help track down the cause

Reply:
From the device itself its not possible, that is the task of the monitoring and control of the workflow.

I have updated the message to be more clear.